arbnode/batch_poster.go

```diff
@@ -1244,6 +1296,12 @@ func (b *BatchPoster) maybePostSequencerBatch(ctx context.Context) (bool, error)
 			b.building.haveUsefulMessage = true
 			break
 		}
+		if config.CheckBatchCorrectness {
+			b.building.muxBackend.allMsgs[b.building.msgCount] = msg
```

Comment:
Since we have CheckBatchCorrectness marked as hot reloadable, `b.building.muxBackend` might be nil here because the config might've been changed to enable checking batch correctness in the middle of batch production. I think the simplest solution is just to make the config not hot reloadable, which I think is fine. It's not a big deal to restart the batch poster to change parameters like that.